bf: S3C-4055 cloudserver metrics computation correction

naren-rajendran opened the pull request:
metrics like incomingBytes, sizeDelta were not calculated correctly for actions like getObject and it is corrected now.

bert-e commented:
# Hello naren-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/cloudserver/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.1
 $ git merge origin/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
```

rahulreddy commented:
@naren-scality Can you please change the base branch to `stabilization/7.9.0`? It needs to be picked up for 7.9.0

miniscruff commented on tests/utapi/utilities.js:
```diff
@@ -0,0 +1,282 @@
+            assert(eventPushed);
+            const compareObj = computeMetrics(event.action, event.metrics);
+            assert(compareObj);
+            Object.keys(compareObj).forEach(key => {
+                assert.strictEqual(eventPushed[key], compareObj[key]);
+            });
```
I think this can be assert.deepStrictEqual instead.

miniscruff commented on tests/utapi/utilities.js:
```diff
@@ -0,0 +1,282 @@
+    },
+}];
+
+function computeMetrics(action, event) {
+    const {
+        numberOfObjects,
+        oldByteLength,
+        newByteLength,
+        byteLength,
+        isDelete,
+    } = event;
+    const incomingBytes = action === 'getObject' ? 0 : newByteLength;
+    const sizeDelta = oldByteLength ? newByteLength - oldByteLength :
+        incomingBytes;
+    return {
+        objectDelta: isDelete ? -numberOfObjects : numberOfObjects,
+        sizeDelta: isDelete ? -byteLength : sizeDelta,
+        incomingBytes,
+        outgoingBytes: action === 'getObject' ? newByteLength : 0,
+    };
+}
```
Duplicating the logic into the test as the code itself isn't the best method of writing tests. For one it ties our tests to the internal logic which we should not do, we just want to test the observable result (eg the output). Additionally, this just proves that what comes in matches what comes out and not that what comes out is correct. For example, if you wrote the test by copy pasting the broken code, the test would still pass even though it was wrong.

Instead, create `expected` objects to compare to in your test array.

Basically it would be the same as doing: 
```js
const jsonString = '{"a": 5}';
assert.strictEqual(JSON.stringify(JSON.parse(jsonString)), jsonString);
```

naren-rajendran replied:
I am still working on this. This is not quite up for review yet, will convert this to draft. I have the correction but did not push it yet... too soon @miniscruff :)

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.1
 $ git merge origin/w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
```

bert-e commented:
# History mismatch



Merge commit #b154ddf694ee11126d8b243bc1ae37be7f71deec on the integration branch
`w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections` is merging a branch which is neither the current
branch `bugfix/S3C-4055-cloudserver-metric-corrections` nor the development branch
`development/7.9`.

It is likely due to a rebase of the branch `bugfix/S3C-4055-cloudserver-metric-corrections` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.1
 $ git merge origin/w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
```

anurag4DSB reviewed (commented):
Do we not have any tests yet?

naren-rajendran commented:
> Do we not have any tests yet?

@anurag4DSB On cloudserver, no, we do not! We have integration tests for utapi v1 but not for v2.

bert-e commented:
# History mismatch



Merge commit #a785c1e9a2761db885ebfaa0a316e4b13f3ff123 on the integration branch
`w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections` is merging a branch which is neither the current
branch `bugfix/S3C-4055-cloudserver-metric-corrections` nor the development branch
`development/7.9`.

It is likely due to a rebase of the branch `bugfix/S3C-4055-cloudserver-metric-corrections` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.1
 $ git merge origin/w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
```

miniscruff commented on tests/utapi/utilities.js:
```diff
@@ -0,0 +1,471 @@
+    testEvents.forEach(event => {
+        it(`should compute and push metrics for ${event.action}`, () => {
+            const eventPushed = pushMetric(event.action, log, event.metrics);
+            assert(eventPushed);
+            Object.keys(event.expected).forEach(key => {
+                assert.deepStrictEqual(eventPushed[key], event.expected[key]);
+            });
```
I think just one assertion will work here.
```suggestion
            assert.deepStrictEqual(eventPushed, event.expected);
```

naren-rajendran replied:
Not all properties of the published event are compared, only metrics.

miniscruff commented on tests/utilities/mock/Utapi.js:
```diff
@@ -0,0 +1,82 @@
+const express = require('express');
```
This feels like a lot of code for a mock. Feels like we would need to add tests here as it has a fair bit of logic.

bert-e commented:
# History mismatch



Merge commit #9d6c6e5eac6fccf689b2e2a40cb48c78c7a4305a on the integration branch
`w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections` is merging a branch which is neither the current
branch `bugfix/S3C-4055-cloudserver-metric-corrections` nor the development branch
`development/7.9`.

It is likely due to a rebase of the branch `bugfix/S3C-4055-cloudserver-metric-corrections` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.1
 $ git merge origin/w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
```

miniscruff commented on tests/utapi/awsNodeSdk.js:
```diff
@@ -0,0 +1,375 @@
+const async = require('async');
```
I feel like this isn't wrong but a bit much. Both with the mock UTAPI server with express and recreating a bunch of events to send to UTAPI. Part of my concern here is too much testing as well as potentially duplicate testing.
It seems to me our unit testing and functional testing overlap here as well as what would probably be included in UTAPI v2. 

I don't really want to block the PR to come up with some sort of test plan or Venn Diagram of what is tested where but I don't want to blindly review all this code.

Commit:
bf: S3C-4055 cloudserver metrics computation correction

metrics like incomingBytes, sizeDelta were not calculated correctly for actions like getObject and it is corrected now.

bert-e commented:
# History mismatch



Merge commit #12b8d17796cf7e27d0a1335ab9c5dcd4146e70cc on the integration branch
`w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections` is merging a branch which is neither the current
branch `bugfix/S3C-4055-cloudserver-metric-corrections` nor the development branch
`development/7.9`.

It is likely due to a rebase of the branch `bugfix/S3C-4055-cloudserver-metric-corrections` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

naren-rajendran commented:
@bert-e approve

bert-e commented:
# History mismatch



Merge commit #12b8d17796cf7e27d0a1335ab9c5dcd4146e70cc on the integration branch
`w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections` is merging a branch which is neither the current
branch `bugfix/S3C-4055-cloudserver-metric-corrections` nor the development branch
`development/7.9`.

It is likely due to a rebase of the branch `bugfix/S3C-4055-cloudserver-metric-corrections` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/7.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/7.9
 $ git merge origin/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.1
 $ git merge origin/w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/7.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/7.9
 $ git merge origin/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.1
 $ git merge origin/w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# History mismatch



Merge commit #58b81aab3b8ff1d61ebdc9ccaf24bd2abe9de79d on the integration branch
`w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` is merging a branch which is neither the current
branch `bugfix/S3C-4055-cloudserver-metric-corrections` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `bugfix/S3C-4055-cloudserver-metric-corrections` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/7.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/7.9
 $ git merge origin/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.1
 $ git merge origin/w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.2`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.2
 $ git merge origin/w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

naren-rajendran commented:
@bert-e help

bert-e commented:
# Help page



The following options and commands are available at this time.

## Options ##


name   | description  | privileged   | authored
------ | ------------ | ------------ |----------
:arrow_right: **create_pull_requests** | Allow the creation of integration pull requests. |  | 
:arrow_right: **bypass_commit_size** | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
:arrow_right: **approve** | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
:arrow_right: **bypass_peer_approval** | Bypass the pull request peers' approval |  :star:  | 
:arrow_right: **after_pull_request** | Wait for the given pull request id to be merged before continuing with the current one. |  | 
:arrow_right: **bypass_leader_approval** | Bypass the pull request leaders' approval |  :star:  | 
:arrow_right: **bypass_build_status** | Bypass the build and test status |  :star:  | 
:arrow_right: **wait** | Instruct Bert-E not to run until further notice. |  | 
:arrow_right: **unanimity** | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
:arrow_right: **bypass_author_approval** | Bypass the pull request author's approval |  :star:  | 
:arrow_right: **bypass_incompatible_branch** | Bypass the check on the source branch prefix |  :star:  | 
:arrow_right: **no_octopus** | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
:arrow_right: **bypass_jira_check** | Bypass the Jira issue check |  :star:  | 



## Commands ##


name   | description  | privileged
------ | ------------ | ------------
:arrow_right: **force_reset** | Delete integration branches & pull requests, and restart merge process from the beginning. | 
:arrow_right: **build** | Re-start a fresh build ```TBA``` | 
:arrow_right: **reset** | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
:arrow_right: **status** | Print Bert-E's current status in the pull request ```TBA``` | 
:arrow_right: **retry** | Re-start a fresh build ```TBA``` | 
:arrow_right: **help** | Print Bert-E's manual in the pull request. | 
:arrow_right: **clear** | Remove all comments from Bert-E from the history ```TBA``` | 






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-4055-cloudserver-metric-corrections` into
`stabilization/7.9.0`
* [w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/7.9...w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/7.9` (pull request #3398)
* [w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/7.10...w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/7.10` (pull request #3399)
* [w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/8.1...w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/8.1` (pull request #3400)
* [w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/8.2...w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/8.2` (pull request #3401)



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **approve**

naren-rajendran commented:
@bert-e status

bert-e commented:
# Status





*Status report is not available.*






*The following options are set:* **approve**

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/7.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/7.9
 $ git merge origin/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.1
 $ git merge origin/w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

naren-rajendran commented:
@bert-e help

bert-e commented:
# Help page



The following options and commands are available at this time.

## Options ##


name   | description  | privileged   | authored
------ | ------------ | ------------ |----------
:arrow_right: **create_pull_requests** | Allow the creation of integration pull requests. |  | 
:arrow_right: **bypass_commit_size** | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
:arrow_right: **approve** | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
:arrow_right: **bypass_peer_approval** | Bypass the pull request peers' approval |  :star:  | 
:arrow_right: **after_pull_request** | Wait for the given pull request id to be merged before continuing with the current one. |  | 
:arrow_right: **bypass_leader_approval** | Bypass the pull request leaders' approval |  :star:  | 
:arrow_right: **bypass_build_status** | Bypass the build and test status |  :star:  | 
:arrow_right: **wait** | Instruct Bert-E not to run until further notice. |  | 
:arrow_right: **unanimity** | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
:arrow_right: **bypass_author_approval** | Bypass the pull request author's approval |  :star:  | 
:arrow_right: **bypass_incompatible_branch** | Bypass the check on the source branch prefix |  :star:  | 
:arrow_right: **no_octopus** | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
:arrow_right: **bypass_jira_check** | Bypass the Jira issue check |  :star:  | 



## Commands ##


name   | description  | privileged
------ | ------------ | ------------
:arrow_right: **force_reset** | Delete integration branches & pull requests, and restart merge process from the beginning. | 
:arrow_right: **build** | Re-start a fresh build ```TBA``` | 
:arrow_right: **reset** | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
:arrow_right: **status** | Print Bert-E's current status in the pull request ```TBA``` | 
:arrow_right: **retry** | Re-start a fresh build ```TBA``` | 
:arrow_right: **help** | Print Bert-E's manual in the pull request. | 
:arrow_right: **clear** | Remove all comments from Bert-E from the history ```TBA``` | 






*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.2`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.2
 $ git merge origin/w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-4055-cloudserver-metric-corrections` into
`stabilization/7.9.0`
* [w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/7.9...w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/7.9` (pull request #3403)
* [w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/7.10...w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/7.10` (pull request #3404)
* [w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/8.1...w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/8.1` (pull request #3405)
* [w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/8.2...w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/8.2` (pull request #3406)



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/c0e1ae18cb04c12cbec15326e5892a7cd3d0991a) did not succeed in branch w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/7b8951b69133414f97a94efc7ea33fbdb01a7b6b) did not succeed in branch w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/c0e1ae18cb04c12cbec15326e5892a7cd3d0991a) did not succeed in branch w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections.




*The following options are set:* **approve**

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/7.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/7.9
 $ git merge origin/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.1
 $ git merge origin/w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.2`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.2
 $ git merge origin/w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-4055-cloudserver-metric-corrections` into
`stabilization/7.9.0`
* [w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/7.9...w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/7.9` (pull request #3407)
* [w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/7.10...w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/7.10` (pull request #3408)
* [w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/8.1...w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/8.1` (pull request #3409)
* [w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/8.2...w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/8.2` (pull request #3410)



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/1078385e20a034e4859219a6f345f02cbf4e79a8) did not succeed in branch w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections.




*The following options are set:* **approve**

bert-e commented:
# History mismatch



Merge commit #1078385e20a034e4859219a6f345f02cbf4e79a8 on the integration branch
`w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections` is merging a branch which is neither the current
branch `bugfix/S3C-4055-cloudserver-metric-corrections` nor the development branch
`development/8.2`.

It is likely due to a rebase of the branch `bugfix/S3C-4055-cloudserver-metric-corrections` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**





*The following options are set:* **approve**

bert-e commented:
# History mismatch



Merge commit #1078385e20a034e4859219a6f345f02cbf4e79a8 on the integration branch
`w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections` is merging a branch which is neither the current
branch `bugfix/S3C-4055-cloudserver-metric-corrections` nor the development branch
`development/8.2`.

It is likely due to a rebase of the branch `bugfix/S3C-4055-cloudserver-metric-corrections` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

naren-rajendran commented:
@bert-e force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/7.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/7.9
 $ git merge origin/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.1
 $ git merge origin/w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.2`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.2
 $ git merge origin/w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-4055-cloudserver-metric-corrections` into
`stabilization/7.9.0`
* [w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/7.9...w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/7.9` (pull request #3411)
* [w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/7.10...w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/7.10` (pull request #3412)
* [w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/8.1...w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/8.1` (pull request #3413)
* [w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/8.2...w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/8.2` (pull request #3414)



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/e30e7356ffb08b6e1c819878708071686f91464c) did not succeed in branch w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections.




*The following options are set:* **approve**

naren-rajendran commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/1063654156f5cdfd4b1ceb3cb262950fcebb8d04) did not succeed in branch w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/a9ff4b04f875688c22150e0b9b89a5c493371e12) did not succeed in branch w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections.




*The following options are set:* **approve**

rahulreddy commented:
/reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**





*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/a9ff4b04f875688c22150e0b9b89a5c493371e12) did not succeed in branch w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections.




*The following options are set:* **approve**

rahulreddy commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/7.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/7.9
 $ git merge origin/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.1
 $ git merge origin/w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.2`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections origin/development/8.2
 $ git merge origin/w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-4055-cloudserver-metric-corrections` into
`stabilization/7.9.0`
* [w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/7.9...w/7.9/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/7.9` (pull request #3415)
* [w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/7.10...w/7.10/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/7.10` (pull request #3416)
* [w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/8.1...w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/8.1` (pull request #3417)
* [w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections](https://github.com/scality/cloudserver/compare/development/8.2...w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections) will be merged into `development/8.2` (pull request #3418)



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/651abc372b9e4c299d8280bcb6bb8995073e4217) did not succeed in branch w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections.




*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections` with contents from `w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections`
and `development/8.2`.

Please resolve the conflict on **the integration branch** (`w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections
 $ git pull  # or "git reset --hard origin/w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections"
 $ git merge origin/development/8.2
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/8.1/bugfix/S3C-4055-cloudserver-metric-corrections
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.2/bugfix/S3C-4055-cloudserver-metric-corrections
```





*The following options are set:* **approve**

rahulreddy commented:
I am going to break 8.1 and 8.2 branches by merging this to allow 7.9 RC4 to be built. @naren-scality  will follow up with a fix for 8.1 and 8.2 branches.

rahulreddy commented:
/bypass_jira_check

rahulreddy commented:
/bypass_build_status

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `stabilization/7.9.0`

* :heavy_check_mark: `development/7.9`

* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/8.1`

* :heavy_check_mark: `development/8.2`



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve, bypass_build_status**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `stabilization/7.9.0`

* :heavy_check_mark: `development/7.9`

* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/8.1`

* :heavy_check_mark: `development/8.2`



The following branches have **NOT** changed:

* `development/6.4`
* `development/7.4`



Please check the status of the associated issue S3C-4055.

Goodbye naren-scality.
